orca-core/src/main/groovy/com/netflix/spinnaker/orca/batch/ExecutionPropagationListener.groovy

```diff
@@ -73,20 +73,24 @@ class ExecutionPropagationListener extends JobExecutionListenerSupport implement
     def id = executionId(jobExecution)
 
     def orcaTaskStatus
-    if (executionRepository.isCanceled(id)) {
-      orcaTaskStatus = CANCELED
-    } else if (jobExecution.failureExceptions) {
+    if (jobExecution.failureExceptions) {
       orcaTaskStatus = TERMINAL
     } else {
       def stepExecutions = new ArrayList<StepExecution>(jobExecution.stepExecutions).sort { it.lastUpdated }.reverse()
       def stepExecution = stepExecutions.find { it.status == jobExecution.status } ?: stepExecutions[0]
-      orcaTaskStatus = stepExecution?.executionContext?.get("orcaTaskStatus") as ExecutionStatus ?: TERMINAL
+      orcaTaskStatus = stepExecution?.executionContext?.get("orcaTaskStatus") as ExecutionStatus
+    }
+
+    if (executionRepository.isCanceled(id) && orcaTaskStatus != TERMINAL) {
```

Comment:
Probably worth adapting one of the `ExecutionPropagationListenerSpec` tests to handle the particular scenario of canceled && TERMINAL == TERMINAL.

Looks like a straight forward test to add.